modules/genai-ecosystem/pages/haystack.adoc

```diff
@@ -0,0 +1,61 @@
+= Haystack Neo4j Integration
+:slug: haystack
+:author: 
+:category: genai-ecosystem
+:tags: haystack, integration, llm, neo4j, vector index
+:neo4j-versions: 5.x
+:page-pagination:
+:page-product: haystack
+
+
+An integration of Neo4j graph database with Haystack v2.0 by deepset. In Neo4j Vector search index is being used for storing document embeddings and dense retrievals.
+
+The library allows using Neo4j as a DocumentStore, and implements the required Protocol methods. You can start working with the implementation by importing it from neo4_haystack package.
+
+== Installation
+
+[source,python]
+----
+# pip install neo4j-haystack
+from neo4j_haystack import Neo4jDocumentStore
+
+document_store = Neo4jDocumentStore(
+    url="bolt://localhost:7687",
+    username="neo4j",
+    password="passw0rd",
+    database="neo4j",
+    embedding_dim=384,
+    embedding_field="embedding",
+    index="document-embeddings", # The name of the Vector Index in Neo4j
+    node_label="Document", # Providing a label to Neo4j nodes which store Documents
+)
+----
+
+== Functionality Includes
+
+* Neo4jEmbeddingRetriever - is a typical retriever component which can be used to query vector store index and find related Documents. The component uses Neo4jDocumentStore to query embeddings.
+* Neo4jDynamicDocumentRetriever is also a retriever component in a sense that it can be used to query Documents in Neo4j. However it is decoupled from Neo4jDocumentStore and allows to run arbitrary Cypher query to extract documents. Practically it is possible to query Neo4j same way Neo4jDocumentStore does, including vector search.
```

Comment:
```
* `Neo4jEmbeddingRetriever`  is a typical retriever component that can be used to query vector store index and find related Documents. The component uses Neo4jDocumentStore to query embeddings.
* `Neo4jDynamicDocumentRetriever` is also a retriever component because it can be used to query Documents in Neo4j. However, it is decoupled from `Neo4jDocumentStore` and allows the running of arbitrary Cypher statements to extract documents. It is possible to query Neo4j the same way `Neo4jDocumentStore` does, including vector search.
```